packages/react-components/theme-designer/src/ThemeDesigner.tsx

```diff
@@ -40,23 +41,68 @@ const useStyles = makeStyles({
 export const ThemeDesigner: React.FC<ThemeDesignerProps> = props => {
   const styles = useStyles();
 
+  const [theme, setTheme] = React.useState<string>('Custom');
   const [keyColor, setKeyColor] = React.useState<string>('#006bc7');
+  const [hueTorsion, setHueTorsion] = React.useState<number>(0);
+  const [darkCp, setDarkCp] = React.useState<number>(2 / 3);
+  const [lightCp, setLightCp] = React.useState<number>(1 / 3);
+
   const debouncedKeyColor = useDebounce(keyColor, 400);
 
-  const themes = React.useMemo(() => {
-    const brand = getBrandTokensFromPalette(debouncedKeyColor);
+  const customTheme = React.useMemo(() => {
+    const brand = getBrandTokensFromPalette(debouncedKeyColor, {
+      hueTorsion: hueTorsion,
+      darkCp: darkCp,
+      lightCp: lightCp,
+    });
     return {
       brand: brand,
       lightTheme: createLightTheme(brand),
       darkTheme: createDarkTheme(brand),
     };
-  }, [debouncedKeyColor]);
+  }, [hueTorsion, darkCp, lightCp, debouncedKeyColor]);
+
+  const themes = React.useMemo(() => {
+    if (theme === 'Teams') {
+      return {
+        brand: brandTeams,
+        lightTheme: createLightTheme(brandTeams),
+        darkTheme: createDarkTheme(brandTeams),
+      };
+    } else if (theme === 'Web') {
+      return {
+        brand: brandWeb,
+        lightTheme: createLightTheme(brandWeb),
+        darkTheme: createDarkTheme(brandWeb),
+      };
+    } else if (theme === 'Office') {
+      return {
+        brand: brandOffice,
+        lightTheme: createLightTheme(brandOffice),
+        darkTheme: createDarkTheme(brandOffice),
+      };
+    } else {
+      return customTheme;
+    }
+  }, [customTheme, theme]);
 
   return (
     <FluentProvider theme={teamsLightTheme}>
       <div className={styles.root}>
         <Nav className={styles.nav} brand={themes.brand} />
-        <Sidebar className={styles.sidebar} keyColor={debouncedKeyColor} setKeyColor={setKeyColor} />
+        <Sidebar
```

Comment:
This is fine for now, but a good exercise after the demo might be to convert all of this complex state into a `useReducer`. We've got a lot of states and setStates here.

Reply:
it would simplify typings as well, as you can define state/reducer once, and simply pass it down. We'll talk about this strategy next week.